### ISSUE TITLE
crypt obfuscator

### ISSUE THREAD

**Issue by oanna500** [UnamSanctam/SilentCryptoMiner#725]
what's better good crypt or obfuscator?

**Comment by UnamSanctam** [comment]
Depends on if you have "Shellcode Loader" enabled or not. A crypter is (usually) for native files (for example C) so you should use that if you have "Shellcode Loader" enabled. An obfuscator is for managed assemblies (for example C#) so you should use that if you have "Shellcode Loader" disabled.

**Comment by oanna500** [comment]
understood, if i use shellcode, `watchdog` and `miner` it is better obfuscate? or is it enough to crypt the final exe

**Comment by UnamSanctam** [comment]
It depends, if you want to decrease (some) runtime detections then you'll most likely need to obfuscate them as well.

**Comment by oanna500** [comment]
how do you think, what's to bypass WD better? `shellcode ON +crypt` or `shellcode OFF +obfuscate`

**Comment by UnamSanctam** [comment]
Right now I believe "Shellcode Loader" disabled with obfuscation is the best against WD (unless you have some good crypter), I even think the miner without "Shellcode Loader" (no obfuscation) is undetected by WD at the moment unless there are some new detections.

**Comment by UnamSanctam** [comment]
You mean any other antivirus than WD? Then it wouldn't really matter, only how good your crypter or obfuscator is would really matter.

**Comment by oanna500** [comment]
understood thanks, I just can't find the best option, shellcode obfuscate crypt. All my tests are not much different from each other. hard to find the best option

**Comment by golxd** [comment]
> Depends on if you have "Shellcode Loader" enabled or not. A crypter is (usually) for native files (for example C) so you should use that if you have "Shellcode Loader" enabled. An obfuscator is for managed assemblies (for example C#) so you should use that if you have "Shellcode Loader" disabled.

if "Shellcode Loader" disabled crypt can't be used? why is better obfuscate?

**Comment by UnamSanctam** [comment]
If "Shellcode Loader" is disabled then the final file is a managed assembly (program made in C#), crypters are mostly for native programs so for a managed program you'd have to use an obfuscator instead. When you have "Shellcode Loader" enabled the "real" miner program made in C# will get converted into shellcode and put into a native C loader that injects the C# program into memory, since the loader is made in C you can crypt it .

**Comment by oanna500** [comment]
sorry) if the miner is on function "Add Defender Exclusions" he first add to Exclusions or will be detected WD

**Comment by UnamSanctam** [comment]
It will add exclusions to Windows Defender on start which stops runtime detections, but scantime detections will still be the same of course.

**Comment by oanna500** [comment]
> If "Shellcode Loader" is disabled then the final file is a managed assembly (program made in C#), crypters are mostly for native programs so for a managed program you'd have to use an obfuscator instead. When you have "Shellcode Loader" enabled the "real" miner program made in C# will get converted into shellcode and put into a native C loader that injects the C# program into memory, since the loader is made in C you can crypt it .

maybe I'm wrong, As I know crypt can C and С#, why you say C# = obfuscate, C=crypt

**Comment by UnamSanctam** [comment]
Because usually crypters only supports native (C) though recently it's been a bit more popular to also support managed (C#).

**Comment by oanna500** [comment]
thanks for your help, i tried test crypt and obfuscate, and I can't figure out which is better. I think main need bypass WD, because many people do not have antivirus, functions Add Defender Exclusions and Disable Windows Update help avoid further detection. Question, where can i check the real detections WD? `antiscan` and `checkzilla` always shows `clean` both crypt so and obfuscate.

**Comment by UnamSanctam** [comment]
You'll have to test it yourself for Windows Defender, there isn't really any other reliable way.

**Comment by oanna500** [comment]
unam i have a question) with update Windows or WD many miners detects, yes you have features "Disable Windows Update" but it doesn't work perfectly, and some computers are still updated. Many say with function "rootkit" miners live longer and not deetct for a long time. Tell me, really rootkit can help with detection WD or antivirus, or not.

**Comment by UnamSanctam** [comment]
No the rootkit generally shouldn't make it more undetected, though the "Disable Windows Update" should work on the majority (if not all) of computers, I haven't been able to make it not work so far since it completely removes the services used for Windows Update.

**Comment by oanna500** [comment]
Windows Update Troubleshooter fixes function "Disable Windows Update" ?

**Comment by oanna500** [comment]
maybe you can help me, I have a laptop with latest working WD, there I test the detections. sometimes, if I copy the miner to a USB flash drive, WD does not see viruses, I scan and everything is fine, but if i upload this miner to any file host (archive with password) and then download it, WD detects a virus. why is this happening. miner file is the same. what is the difference between download and usb flash, maybe you know more than me

**Comment by UnamSanctam** [comment]
Some file hosts (like MediaFire) and others scan the file with VirusTotal which can cause that. Does it happen with hosts like Anonfiles? Windows Defender does also scan downloads more thoroughly, does it get detected before you extract it or after?

**Comment by oanna500** [comment]
after extract (host Mega.nz) but archive have password. strange

**Comment by oanna500** [comment]
Unam hello, please tell me, we sometimes chat in telegram, some say miner may not start after reboot, and that the miner is not stable when using CPU+GPU (example xmr+etc). Everything is fine on my computer, just wanted to ask.
did you have any problems in the tests?

**Comment by UnamSanctam** [comment]
No I haven't had any problems, and I've never received any reports of it where it's not due to something the user has done.

**Comment by oanna500** [comment]
Unam, maybe a stupid question, i saw your config .crproj for confuser, it has 8 positions
```
<protection id="constants" />
    <protection id="ctrl flow" />
    <protection id="harden" />
    <protection id="invalid metadata" />
    <protection id="ref proxy" />
    <protection id="rename" />
    <protection id="typescramble" />
    <protection id="anti debug" />
```
what if i use only `constants `or `ref proxy` or any other, and I have fewer detections, the question is what)
for the work of the miner anyway how many protections `1` or `10`, or the less the better, or doesn't matter, or only need look at the amount of detections

**Comment by UnamSanctam** [comment]
Only the detections really matter (and make sure everything works after obfuscation).

**Comment by oanna500** [comment]
understood thanks. Miner usually works, other functions can not check "Disable Windows Update" "Disable Sleep" "Block Websites" etc. I hope they work too)

**Comment by UnamSanctam** [comment]
Yeah those options should work fine without needing to test them, they are only commands that gets run in Windows (except "Block Websites" which just writes text into the hosts file), and even if the commands don't work it shouldn't affect the miner since they are running through CMD.

**Comment by oanna500** [comment]
sorry) I check the miner to detect WD in windows 11 pro, you do not know, where is the best place to do it, on which version windows 7 10 11? home professional corporate? Is WD somehow different?

**Comment by UnamSanctam** [comment]
Windows Defender should be relatively the same on versions of Windows 10 and 11 so any there should work. For Windows 7 the Windows Defender there barely works so you shouldn't try that.

**Comment by oanna500** [comment]
unfortunately( tried different obfuscators... when starting the miner, WD blocks the miner, virus `Wacatac `or `Sabsik`, but I looked at WD exclusions, and there was disk C on exclusions. did he not have time to add himself to exclusions? or how this function works. Tried 1-2 weeks ago, wd didn't block miner, although WD works differently on different computers.

**Comment by UnamSanctam** [comment]
Windows Defender won't block the miner if you run it on the C: disk while the exclusions are there, show a screenshot of the detection.

**Comment by UnamSanctam** [comment]
Well yes, it blocks/removes it before it installs from what it sounds like, meaning that your miner is getting file scan detections.

**Comment by oanna500** [comment]
I just don't understand why exclusions is on C disk, miner is blocked, probably and exclusions shouldn't have worked?

**Comment by UnamSanctam** [comment]
You just said that you are running it on the D: drive, meaning the C: drive won't affect it, it will only avoid detections on the C: drive when installed but you are probably getting detections from before it gets installed, you can check the details of the detection to see what it detects.

**Comment by oanna500** [comment]
tried start miner from drive C, WD didn't even let it run. 
![Untitled](https://user-images.githubusercontent.com/105070556/174410764-1e654141-7d6d-4926-9b40-7a3f38c9b667.png)
it starts from drive D, but WD blocks it too.
ok, no more questions) looks like there's nothing I can do about it

**Comment by UnamSanctam** [comment]
Yes that's correct.
